### PR TITLE
Added deleteTemplate method

### PR DESCRIPTION
**Pull request by CCphobia**
#4 Added deleteTemplate method in repo and service and wrote respective test methods.
Adjusted testAddTemplate to avoid problems with hibernate having problems with @ lob object.
Renamed testGetTemplateShouldThrowNoSuchEntityExceptioan() to properly represent tested problem

### CODE REVIEW

File: src/main/java/pwr/asi/java/hermes/services/TemplateService.java [hunk 1]
```diff
@@ -39,4 +39,14 @@ public Template getTemplate(String title) throws NoSuchEntityInDBException {
         return templateRepository.getByTitle(title)
                 .orElseThrow(() -> new NoSuchEntityInDBException("No such entity in database"));
     }
+
+    @Transactional
+    public void deleteTemplateByTitle(String title) throws NoSuchEntityInDBException {
+        var template = templateRepository.getByTitle(title);
+        if (template.isPresent()) {
```
Comment (Burdzi0) [inline src/main/java/pwr/asi/java/hermes/services/TemplateService.java]: Use `ifPresentOrElse`

Comment (Burdzi0) [inline src/main/java/pwr/asi/java/hermes/services/TemplateService.java]: Please use `ifPresentOrElse()`

**Review by KarolKalbarczyk — COMMENTED** [review]
in testDeleteTemplateByTitle add assert throw for trying to get Template "A"

File: src/test/java/pwr/asi/java/hermes/services/TemplateServiceTest.java [hunk 2]
```diff
@@ -35,7 +35,7 @@ void testAddTemplate() {
         templateService.addTemplate(template);
 
         assertThat(templateRepository.count()).isEqualTo(1);
-        assertThat(templateRepository.getByTitle(template.getTitle())).isNotEmpty();
+        assertThat(templateService.getTemplate(template.getTitle()).equals(template));
```
Comment (Burdzi0) [inline src/test/java/pwr/asi/java/hermes/services/TemplateServiceTest.java:39]: Use `isEqualTo` as above or change the above to use `equals` as here

File: src/main/java/pwr/asi/java/hermes/services/TemplateService.java [hunk 3]
```diff
@@ -39,4 +39,14 @@ public Template getTemplate(String title) throws NoSuchEntityInDBException {
         return templateRepository.getByTitle(title)
                 .orElseThrow(() -> new NoSuchEntityInDBException("No such entity in database"));
     }
+
+    @Transactional
+    public void deleteTemplateByTitle(String title) throws NoSuchEntityInDBException {
+        var template = templateRepository.getByTitle(title);
```
Comment (Burdzi0) [inline src/main/java/pwr/asi/java/hermes/services/TemplateService.java]: Maybe would it be nice to use `final`?

**Review by Burdzi0 — CHANGES_REQUESTED** [review]
Change to `ifPresentOrElse()`, the rest is up to You

**Comment by Burdzi0** [comment]
Also read: https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords
I'm pretty sure You need some keyword before the issue number to close it

**Review by Burdzi0 — APPROVED** [review]

**Pull request merged**